Description: migrate to SDL2 but no sound
 Migrate from SDL1.2 to SDL2, but lost the sound
so sadly, this patch is not activated
Author: Nicolas HOUDELOT <nicolas@demosdebs.org>
Applied-Upstream: no
Last-Update: 2024-03-18
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/main.c
+++ b/main.c
@@ -1,7 +1,7 @@
 
 #include <string.h>
 #include <stdlib.h>
-#include <SDL/SDL.h>
+#include <SDL2/SDL.h>
 #include <math.h>
 #ifdef MAC
 #include <gl.h>
@@ -37,8 +37,8 @@
 int main(int argc, char *argv[])
 {
 	int i,quit=0,xs=XS,ys=YS,flagga=0,aa=1,param=0;
-	SDL_Event e;
 	SDL_AudioSpec w;
+	SDL_Event e;
 
 	SDL_Init(SDL_INIT_VIDEO|SDL_INIT_TIMER|SDL_INIT_AUDIO);
 
@@ -53,8 +53,11 @@
 #endif
 	w.callback=pakki;
 	w.userdata=NULL;
-	SDL_OpenAudio(&w,NULL);
+	if (SDL_OpenAudio(&w,NULL)>0)
+		return(EXIT_FAILURE);
+
 	syna_init(44100);
+
 	if(syna_get(piisi)<0)
 		return(EXIT_FAILURE);
 
@@ -62,7 +65,7 @@
 	{
 		if(!strcmp(argv[i],"-w"))
 		{
-			flagga-=SDL_FULLSCREEN;
+			flagga-=SDL_WINDOW_FULLSCREEN;
 			param++;
 		}
 		if(!strcmp(argv[i],"-a"))
@@ -73,7 +76,7 @@
 	}
 
 	if(argv[0][strlen(argv[0]-1)]=='w')
-		flagga-=SDL_FULLSCREEN;
+		flagga-=SDL_WINDOW_FULLSCREEN;
 
 	if(argc>=param+3)
 	{
@@ -85,17 +88,20 @@
 	if(aa && !getenv("__GL_FSAA_MODE"))
 		putenv("__GL_FSAA_MODE=4");
 
-	flagga+=SDL_OPENGL|SDL_FULLSCREEN;
+	flagga+=SDL_WINDOW_FULLSCREEN_DESKTOP;
 	if(argv[0][strlen(argv[0])-1]=='w')
-		flagga-=SDL_FULLSCREEN;
+		flagga-=SDL_WINDOW_FULLSCREEN_DESKTOP;
 
 #ifdef MAC
 	SDL_SetVideoMode(xs,ys,32,flagga);
 #else
 	//SDL_SetVideoMode(xs,ys,32,SDL_DOUBLEBUF|SDL_OPENGL|SDL_FULLSCREEN);
-	SDL_SetVideoMode(xs,ys,32,SDL_DOUBLEBUF|flagga);
+	SDL_Window *sdlWindow;
+	SDL_GLContext sdlGlcontext;
+	sdlWindow = SDL_CreateWindow("Leeko by Fit",SDL_WINDOWPOS_CENTERED,SDL_WINDOWPOS_CENTERED,xs,ys,SDL_WINDOW_OPENGL|flagga);
+	sdlGlcontext = SDL_GL_CreateContext(sdlWindow);
 #endif
-	SDL_WM_SetCaption("Leeko",0);
+
 	SDL_ShowCursor(0);
 
 	lego_init();
@@ -107,7 +113,6 @@
 	writer_init();
 	naamat_init();
 	SDL_PauseAudio(0);
-
 	while(!quit)
 	{
 		glClearColor(0,0,0,0);
@@ -122,7 +127,7 @@
 		if(beat>=73)
 			quit=1;
 
-		SDL_GL_SwapBuffers();
+		SDL_GL_SwapWindow(sdlWindow);
 		while(SDL_PollEvent(&e)>0)
 		{
 			if(e.type==SDL_MOUSEBUTTONDOWN)
@@ -135,7 +140,8 @@
 		}
 	}
 
-	SDL_Quit();
+	SDL_GL_DeleteContext(sdlGlcontext);
+	SDL_DestroyWindow(sdlWindow);
 	return(EXIT_SUCCESS);
 }
 
--- a/Makefile
+++ b/Makefile
@@ -4,14 +4,14 @@
 
 CC := gcc
 PKG_CONFIG:=pkg-config
-CFLAGS += -Wall -pipe -fno-strict-aliasing -std=gnu90 $(shell $(PKG_CONFIG) --cflags sdl)
+CFLAGS += -Wall -pipe -fno-strict-aliasing -std=gnu90 $(shell $(PKG_CONFIG) --cflags sdl2)
 HEADERS := palikka.h volume.h gl_render.h dump.h wavefront.h dudes.h \
           hakkyra.h tunneli.h spiral.h ball.h writer.h feidit.h naamat.h \
           syna.h
 OBJ := palikka.o volume.o main.o wavefront.o dump.o gl_render.o dudes.o \
       hakkyra.o tunneli.o spiral.o serfdata.o walkerdata.o ball.o fontdata.o \
       writer.o feidit.o naamatdata.o naamat.o syna.o piisi.o
-LDFLAGS += $(shell $(PKG_CONFIG) --libs sdl) -lm
+LDFLAGS += $(shell $(PKG_CONFIG) --libs sdl2) -lm
 TARGET := leeko-by-fit
 
 INSTALL := install
